File: src/test/java/net/haesleinhuepf/clij/ops/examples/TopHat.java

package net.haesleinhuepf.clij.ops.examples;

import java.io.IOException;

import org.junit.Test;

import net.haesleinhuepf.clij.ops.CLIJ_close.CLIJ_close;
import net.haesleinhuepf.clij.ops.CLIJ_maximumBox.CLIJ_maximumBox;
import net.haesleinhuepf.clij.ops.CLIJ_medianSliceBySliceSphere.CLIJ_medianSliceBySliceSphere;
import net.haesleinhuepf.clij.ops.CLIJ_minimumBox.CLIJ_minimumBox;
import net.haesleinhuepf.clij.ops.CLIJ_pull.CLIJ_pull;
import net.haesleinhuepf.clij.ops.CLIJ_push.CLIJ_push;
import net.haesleinhuepf.clij.ops.CLIJ_subtractImages.CLIJ_subtractImages;
import net.imagej.ImageJ;

public class TopHat {

	@Test
	public void run() throws IOException {
		int radiusXY = 10;
		int radiusZ = 0;
		ImageJ ij = new ImageJ();
		Object input = ij.io().open(getClass().getResource("/samples/t1-head.tif").getPath());
		ij.ui().show("input", input);
		// push image to GPU
		Object inputGPU = ij.op().run(CLIJ_push.class, input);
		// median slice by slice
		Object median = ij.op().run(CLIJ_medianSliceBySliceSphere.class, inputGPU,
			1, 1);
		Object temp1 = ij.op().run(CLIJ_minimumBox.class, median, radiusXY,
			radiusXY, radiusZ);
		Object temp2 = ij.op().run(CLIJ_maximumBox.class, temp1, radiusXY, radiusXY,
			radiusZ);
		Object output = ij.op().run(CLIJ_subtractImages.class, median, temp2);

		// show result
		Object result = ij.op().run(CLIJ_pull.class, output);
		ij.ui().show("top hat", result);

		// cleanup
		ij.op().run(CLIJ_close.class, inputGPU);
		ij.op().run(CLIJ_close.class, median);
		ij.op().run(CLIJ_close.class, temp1);
		ij.op().run(CLIJ_close.class, temp2);
		ij.op().run(CLIJ_close.class, output);
		ij.op().run(CLIJ_close.class);

	}

	public static void main(String... args) throws IOException {
		TopHat task = new TopHat();
		task.run();
	}
}

File: src/test/resources/jython/topHat.py
# CLIJ example ImageJ Ops jython: topHat.py
#
# This script shows how a top-hat filter can be applied in the GPU.
#
# Author: Robert Haase, rhaase@mpi-cbg.de
# Author: Deborah Schmidt, frauzufall@mpi-cbg.de
# May 2019
# ---------------------------------------------

#@ IOService io
#@ UIService ui
#@ OpService ops

radiusXY = 10
radiusZ = 0

import os
import inspect
def getResource(file):
    return os.path.dirname(os.path.abspath(inspect.getsourcefile(lambda:0))) + "/" + file


input = io.open(getResource("../samples/t1-head.tif"))
ui.show("input", input)
# push image to GPU
inputGPU = ops.run("CLIJ_push", input)
median = ops.run("CLIJ_medianSliceBySliceSphere", inputGPU, 1, 1)
temp1 = ops.run("CLIJ_minimumBox", median, radiusXY, radiusXY, radiusZ)
temp2 = ops.run("CLIJ_maximumBox", temp1, radiusXY, radiusXY, radiusZ)
output = ops.run("CLIJ_subtractImages", median, temp2)
# show result
result = ops.run("CLIJ_pull", output)
ui.show("top hat", result)

#cleanup
ops.run("CLIJ_close", inputGPU)
ops.run("CLIJ_close", median)
ops.run("CLIJ_close", temp1)
ops.run("CLIJ_close", temp2)
ops.run("CLIJ_close", output)
ops.run("CLIJ_close")
